### ISSUE TITLE
AppImage fail to start correctly

### ISSUE THREAD

**Issue by flux242** [marcogrecopriolo/guglielmo#14]
Missing libs for the HW. These libs should be inside of the AppImage but they aren't
```
./guglielmo-x86_64-v0.4.dab.receiver.AppImage 
error report libsdrplay_api.so: cannot open shared object file: No such file or directory
we could not load libmirsdrapi-rsp
If that is not the device you were expecting, do not worry
failed to open librtlsdr.so
Error = librtlsdr.so: cannot open shared object file: No such file or directory
failed to open libairspy.so
Error = libairspy.so: cannot open shared object file: No such file or directory
failed to open libLimeSuite.so
failed to open libhackrf.so
Segmentation fault
```

**Comment by marcogrecopriolo** [comment]
None of welle-io, Qt-Dab and guglielmo ship packages for the devices with the AppImage.
For most the devices the issue is that it's not just a case of shipping a single library, you require a package with several libraries and ancillary files.
Librtlsdr.so might be easy enough, however you will find that in most cases, you require libraries built on your particular distro because they rely on specific symbol versions from other libraries, which again are peculiar to your distro.
Some, like sdrplay, are not even shipped with your distro, and either are prebuilt (ubuntu / fedora), or require building on your own box. 

In short, you are expected to install the package for the device that you use, from your own distro.

What I am more concerned is the segmentation fault, which I have not seen.
Could you run the app image from gdb and pass me a stack?

**Comment by flux242** [comment]
of course they ship librtlsdr and libairspy shared lib inside of their AppImages because otherwise it wouldn't work. So should you

**Comment by flux242** [comment]
btw
```
dpkg -L librtlsdr0
/.
/etc
/etc/modprobe.d
/lib
/lib/udev
/lib/udev/rules.d
/lib/udev/rules.d/60-librtlsdr0.rules
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/librtlsdr.so.0.6.0
/usr/share
/usr/share/doc
/usr/share/doc/librtlsdr0
/usr/share/doc/librtlsdr0/changelog.Debian.gz
/usr/share/doc/librtlsdr0/copyright
/usr/share/metainfo
/usr/share/metainfo/librtlsdr0.metainfo.xml
/usr/lib/x86_64-linux-gnu/librtlsdr.so.0
```

**Comment by flux242** [comment]
gqrx also ships hackrf lib

**Comment by flux242** [comment]
as you wish. I'm fine with well-io

**Comment by Tractor777** [comment]
I have a similar problem; when I run the AppImage it starts but finds no devices (errors as above minus the segmentation fault). I have the RTLSDR device installed as GQRX and Well.io Appiamages both work fine. I have the Librtlsdr installed on Ubuntu 2204. Can you provide any help on what needs to be installed to get your software to work? Otherwise, I'll have to stick with well-io which works fine.

**Comment by marcogrecopriolo** [comment]
As you see from the message, the appimage is looking for librtlsdr.so.
From

https://packages.debian.org/search?searchon=contents&keywords=librtlsdr.so

On Ubuntu you need to install librtlsdr-dev (I guess :amd64 in your case)
For rpm based distributions

https://rpmfind.net/linux/rpm2html/search.php?query=librtlsdr.so

it's called rtl-sdr-devel-xxx.rpm, where xxx is the version

**Comment by Tractor777** [comment]
Many thanks it was this package that was missing librtlsdr-dev . I can now recieve DAB (for reference, I already had librtlsdr0 installed, but needed the librtlsdr-dev as well)

**Comment by flux242** [comment]
-dev package dependency to start an application is a no go

**Comment by marcogrecopriolo** [comment]
I don't control in which package distributions place libraries, and it's even more stupid considering that for the debian package, it's just a link to libtlsdr.so.0.6 that they are creating.
It should have been part of librtlsdr0, but there you go.

**Comment by marcogrecopriolo** [comment]
The next appimage will contain the rtlsdr shared library.
To be clear - this still doesn't help you and you still have to install the required packages, because in order for the library to be able to open the device, you need to create udev rules, restart udev, and create an rtlsdr user, all of which is achieved by installing the package that ships the library in the first place.
Please don't say that you expect me to do that too.
